### DISCUSSION TITLE
Keycloak brokering: Auth0

### DISCUSSION THREAD

**Discussion by malejpavouk** [keycloak/keycloak#12198] [Q&A]
Hi,
we are trying to make brokering work with Auth0 (as the remote IdP). We have encountered the issue that Auth0 by default returns opaque JWT tokens, so we can't use token mappers in KC (since the claims are not in the access token).
Documentation of Auth0 indicates that the client (in this case KC), should send an audience parameter to switch to a regular JWT token (on Auth0 side). But it does not seem that there is such a switch on brokering page of Keycloak.

Does anyone have any idea how to resolve this issue?

https://auth0.com/docs/secure/tokens/access-tokens/get-access-tokens#example-post-to-token-url

**Comment by pedroigor** [comment]
I guess you are using an OpenID Connect Broker.

If so, this broker is able to obtain user information from the User Info Endpoint if enabled.

Or Auth0 does not return an `id_token`?

**Reply by malejpavouk** [reply]
yes, OpenId broker (sorry for missing this info in the question). The ID token works fine, it's always in an open format (normal base64(JSON)). Only the access token is opaque in case the audience parameter is not passed with the token request.

**Comment by MurariJha** [comment]
After some effort, I discovered a solution. You can append it to the Identity providers authorization URL as shown below.
{OAuth Authorization url}/authorize?audience={audience url}
